src/test/java/it/eng/idsa/dataapp/handler/DescriptionRequestMessageHandlerTest.java

```diff
@@ -0,0 +1,75 @@
+package it.eng.idsa.dataapp.handler;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.when;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.apache.commons.lang3.StringUtils;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.http.ResponseEntity;
+import org.springframework.test.util.ReflectionTestUtils;
+import org.springframework.web.client.RestTemplate;
+
+import de.fraunhofer.iais.eis.Connector;
+import de.fraunhofer.iais.eis.Message;
+import de.fraunhofer.iais.eis.ids.jsonld.Serializer;
+import it.eng.idsa.dataapp.configuration.ECCProperties;
+import it.eng.idsa.multipart.processor.util.SelfDescriptionUtil;
+import it.eng.idsa.multipart.util.UtilMessageService;
+
+class DescriptionRequestMessageHandlerTest {
+
+	@InjectMocks
+	private DescriptionRequestMessageHandler descriptionRequestMessageHandler;
+	@Mock
+	private RestTemplate restTemplate;
+	@Mock
+	private RestTemplateBuilder restTemplateBuilder;
+	@Mock
+	private ECCProperties eccProperties;
+	@Mock
+	private ResponseEntity<String> response;
+	private Message message;
+	Map<String, Object> responseMap = new HashMap<>();
+	private Serializer serializer = new Serializer();
+	private Connector baseConnector;
+	private String issuerConnector = "http://w3id.org/engrd/connector/";
+
+	@BeforeEach
+	public void init() throws IOException, URISyntaxException {
+
+		MockitoAnnotations.initMocks(this);
+		ReflectionTestUtils.setField(descriptionRequestMessageHandler, "restTemplate", restTemplate);
```

Comment:
Why ReflectionTestUtils for RestTemplate?
Can we create
@Mock 
RestTemplate restTemplate
and then do
when(RestTemplateBuilder.build()).thenReturn(restTemplate)?